### PR TITLE
expression: implement vectorized evaluation for `builtinCastDecimalAsDurationSig`

### PR DESCRIPTION
**Pull request by dengsigua**
### What problem does this PR solve?
implement vectorized evaluation for builtinCastDecimalAsDurationSig, for [#12106](https://github.com/pingcap/tidb/issues/12106)

### What is changed and how it works?
according to benchmark, about 5 times faster than before:
```
BenchmarkVectorizedBuiltinCastFunc/builtinCastDecimalAsDurationSig-VecBuiltinFunc-8               366718              3352 ns/op              0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinCastFunc/builtinCastDecimalAsDurationSig-NonVecBuiltinFunc-8             77730             15277 ns/op              0 B/op          0 allocs/op     
```
### Check List 

Tests 

 - Unit test\

### CODE REVIEW

**Comment by francis0407** [comment]
Please run `go fmt` in the expression package.

**Comment by dengsigua** [comment]
> Please run `go fmt` in the expression package.



> Please run `go fmt` in the expression package.

already modified, plz check
thanks~

File: expression/builtin_cast_vec.go [hunk 1]
```diff
@@ -1509,11 +1509,44 @@ func (b *builtinCastDecimalAsIntSig) vecEvalInt(input *chunk.Chunk, result *chun
...
+			if types.ErrTruncatedWrongVal.Equal(err) {
+				err = b.ctx.GetSessionVars().StmtCtx.HandleTruncate(err)
+			}
+			if err != nil {
+				return err
+			}
+			if dur == types.ZeroDuration {
+				result.SetNull(i, true)
+			}
+			continue
```
Comment (qw4990) [inline expression/builtin_cast_vec.go]: This `continue` should be in line 1544.
If the error is handled and the `dur` is not equal to `ZeroDuration`, it is wrong.

**Pull request merged**